README.md

```diff
@@ -5,3 +5,44 @@ Adds automated code upgrades to REDAXO improving developer productivity and code
 The AddOn integrates [rector](https://github.com/rectorphp/rector) with the developer in mind, meaning it eases use for often used migration/upgrade use-cases. If you are a rector expert you don't need this AddOn. 
 
 Primary purpose is to allow people less experienced with developer tooling to automate migration tasks.
+
+Users apply the changes after controlling the changes in a diff view.
+
+## use cases
+
+### PHP Version Migrations
+
+This use case helps updating the PHP version used in a project to a newer one. This can involve migrating code to be compatible with the new version and updating any deprecated features to the recommended replacements.
+
+Select the desired php version to start migrating.
+
+### Improve Code Quality
+
+- `Unify Code Quality:` In this use case, the goal is to ensure that all code in a project adheres to the same coding standards and best practices.
+
+- `Remove Dead Code:` This use case involves identifying and removing code that is no longer used or needed in the codebase.
+
+- `Infer Type Declarations:` In this use case, the goal is to automatically infer the types of variables and functions in the codebase.
+
+- `Reduce Symbol Visibility (Privatization):` This use case involves reducing the visibility of symbols in the codebase to make it more modular and easier to understand. 
+
+-  `Use Early Returns:` In this use case, the goal is to reduce the complexity of code by using early returns to exit a function as soon as possible if a certain condition is met. 
+
+### PHPUnit Version Migrations: 
+
+This use cases involve updating the version of PHPUnit used in a project to a newer one. This can involve migrating test code to be compatible with the new version and updating any deprecated features to the recommended replacements.
+
+### Improve Test-Code Quality: 
+
+This use cases involve improving the quality of test code by making it more maintainable, readable, and efficient. This can include refactoring existing test code to follow best practices, removing duplication, and improving the structure of test suites.iently. 
+
+### Misc 
+
+- `REDAXO Specific Code Style:` This use case involves ensuring that code written for REDAXO follows the established code style guidelines for the platform. This can include adhering to naming conventions, using appropriate formatting, and following the coding standards recommended by the REDAXO community.
```

Comment:
```suggestion
- `REDAXO Specific Code Style:` This use case ensures that code written follows the [REDAXO code style guidelines](https://github.com/redaxo/php-cs-fixer-config). This can include adhering to naming conventions, using appropriate formatting, and following the coding standards recommended by the REDAXO community.
```